### DISCUSSION TITLE
EDIT "the_author_posts_link"

### DISCUSSION THREAD

**Discussion by avdsm** [bootscore/bootscore#249] [Q&A] [answered]
Good afternoon
I had to remove the update date of the post, and the image before the link to the author's profile.
I covered it with CSS.
Is there another way to be able to not only hide, but also change the icon in front of the author?
example here: https://vosmarket.am/recipes/havic-apur-manukneri/

**Comment by crftwrk** [comment, marked as answer]
Hello,

this are all pluggable functions in https://github.com/bootscore/bootscore/blob/main/inc/template-tags.php. You can edit all by copy the function between the `if (!function_exists('bootscore_function_name')) :` ... `endif;` to your child's `functions.php`.

For example this one removes the date updated:

```php
function bootscore_date() {
  $time_string = '<time class="entry-date published updated" datetime="%1$s">%2$s</time>';
  if (get_the_time('U') !== get_the_modified_time('U')) {
    $time_string = '<time class="entry-date published" datetime="%1$s">%2$s</time>';
  }

  $time_string = sprintf(
    $time_string,
    esc_attr(get_the_date(DATE_W3C)),
    esc_html(get_the_date())
  );

  $posted_on = sprintf(
    /* translators: %s: post date. */
    '%s',
    '<span rel="bookmark">' . $time_string . '</span>'
  );

  echo '<span class="posted-on">' . $posted_on . '</span>'; // WPCS: XSS OK.

}
```

This one adds a icon to the user in loop:

```php
function bootscore_author() {
  $byline = sprintf(
    esc_html_x('by %s', 'post author', 'bootscore'),
    '<span class="author vcard"><i class="fas fa-user"></i>&nbsp;<a class="url fn n" href="' . esc_url(get_author_posts_url(get_the_author_meta('ID'))) . '">' . esc_html(get_the_author()) . '</a></span>'
  );

  echo '<span class="byline"> ' . $byline . '</span>'; // WPCS: XSS OK.

}

```

But of course, you're right. In `single.php` is `the_author_posts_link();` used instead `bootscore_author();`. I do not know why it's still there, we will change this in one of the next releases. Thank you for pointing this out.

However, just open `single.php` in your child and search for:

```php
<?php
bootscore_date();
_e(' by ', 'bootscore');
the_author_posts_link();
bootscore_comment_count();
?>
```

Replace with this:

```php
<?php
bootscore_date();
bootscore_author();
bootscore_comment_count();
?>
```

Does this help?

**Reply by avdsm** [reply]
Thanks

**Reply by avdsm** [reply]
After adding these features, they disappeared translating "Leave a comment" on the recipes and posts page, "Tagged" on the post page and blog categories page.

**Reply by crftwrk** [reply]
> After adding these features, they disappeared translating "Leave a comment" on the recipes and posts page, "Tagged" on the post page and blog categories page.

This affects only date/author and not comments/tags. I triple checked it with different languages and they work fine. So, I cannot explain what happened on your site. Is it possible that your translation is not 100% complete or maybe it's one of your plugins (server-side caching etc.)?

However, this seems to be a good point to add your translation to the theme. If you want to contribute, simply fork the repo, add your translation and create a pull request. We can then further check the missing strings.

`single-*.php`'s are updated here https://github.com/bootscore/bootscore/pull/251

**Reply by avdsm** [reply]
Good! I will finish the translation and write to you.

**Reply by avdsm** [reply]
> > After adding these features, they disappeared translating "Leave a comment" on the recipes and posts page, "Tagged" on the post page and blog categories page.
.....
> However, this seems to be a good point to add your translation to the theme. If you want to contribute, simply fork the repo, add your translation and create a pull request. We can then further check the missing strings.
> 
> `single-*.php`'s are updated here #251

I translated everything into Armenian and everything works fine.
How to send you a transfer?

**Reply by avdsm** [reply]
Not a "transfer", but a translation into Armenian!

**Reply by crftwrk** [reply]
Great! Simply press the "Fork" button on the top of this page. Then go to your fork theme folder "languages" and click "Add file" > "Upload files". Choose your .mo and .po translation and upload them, press "Commit changes". Then press "Contribute" > "Open pull request" on the top. This is the easiest way.
